pkg/cmd/cli/cmd/loginoptions.go

```diff
@@ -92,17 +98,82 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 		}
 	}
 
-	// normalize the provided server to a format expected by config
-	serverNormalized, err := config.NormalizeServerURL(o.Server)
-	if err != nil {
+	// completeClientConfig checks if the provided host is reachable and if so, completes the clientConfig with
+	// CA data and other information required to make connections to the given server
+	if err := o.completeClientConfig(clientConfig); err != nil {
 		return nil, err
 	}
-	o.Server = serverNormalized
+	o.Config = clientConfig
+
+	return o.Config, nil
+}
+
+func (o *LoginOptions) completeClientConfig(clientConfig *restclient.Config) error {
+	if err := o.normalizeAndDialToServer(); err != nil {
```

Comment:
why mix normalization and dialing? keeping them separate lets us assign clientConfig.Host once, and test normalization separately